src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1053,6 +969,86 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		}
 	}
 
+	shouldProcessKeyEvent(event: KeyboardEvent): boolean {
+		// Disable all input if the terminal is exiting
+		if (this._isExiting) {
+			return false;
+		}
+
+		const standardKeyboardEvent = new StandardKeyboardEvent(event);
+		const resolveResult = this._keybindingService.softDispatch(standardKeyboardEvent, standardKeyboardEvent.target);
+
+		// Respect chords if the allowChords setting is set and it's not Escape. Escape is
+		// handled specially for Zen Mode's Escape, Escape chord, plus it's important in
+		// terminals generally
+		const isValidChord = resolveResult.kind === ResultKind.MoreChordsNeeded && this._configHelper.config.allowChords && event.key !== 'Escape';
+		if (this._keybindingService.inChordMode || isValidChord) {
+			event.preventDefault();
+			return false;
+		}
+
+		const SHOW_TERMINAL_CONFIG_PROMPT_KEY = 'terminal.integrated.showTerminalConfigPrompt';
+		const EXCLUDED_KEYS = ['RightArrow', 'LeftArrow', 'UpArrow', 'DownArrow', 'Space', 'Meta', 'Control', 'Shift', 'Alt', '', 'Delete', 'Backspace', 'Tab'];
+
+		// only keep track of input if prompt hasn't already been shown
+		if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT_KEY, StorageScope.APPLICATION, true) &&
+			!EXCLUDED_KEYS.includes(event.key) &&
+			!event.ctrlKey &&
+			!event.shiftKey &&
+			!event.altKey) {
+			this._hasHadInput = true;
+		}
```

Comment:
This has a bunch of side effects so exposing it, especially with the name `shouldProcessKeyEvent` is misleading. I'm also a little worried about exposing such an internal function on `ITerminalInstance`. Is `isSingleLetterKey` good enough to catch most cases?

Reply:
Hmm, I will explore some more. I was mostly concerned with ensuring the accessible view isn't exited inappropriately. It seemed like I'd be duplicating a lot of this code, which is why I exposed it.